framework/src/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java

```diff
@@ -891,6 +1029,34 @@ private AnnotationMirror createNumberAnnotationMirror(List<Number> values) {
                 "ValueAnnotatedTypeFactory: unexpected class: " + first.getClass());
     }
 
+    public AnnotationMirror createIntRangeAnnotation(Range range) {
+        if (range.from > range.to || range.from == Long.MIN_VALUE && range.to == Long.MAX_VALUE) {
+            return UNKNOWNVAL;
+        } else if (!range.isWiderThan(MAX_VALUES)) {
+            List<Long> values = new ArrayList<>();
+            for (long value = range.from; value <= range.to; value++) {
+                values.add(value);
+            }
+            return createIntValAnnotation(values);
+        }
+        AnnotationBuilder builder = new AnnotationBuilder(processingEnv, IntRange.class);
```

Comment:
Agree. It's the original code's way of handling this and I didn't want to make it inconsistent. Will refactor